numpy/f2py/crackfortran.py

```diff
@@ -869,6 +869,8 @@ def appenddecl(decl, decl2, force=1):
 
 selectpattern = re.compile(
     r'\s*(?P<this>(@\(@.*?@\)@|[*][\d*]+|[*]\s*@\(@.*?@\)@|))(?P<after>.*)\Z', re.I)
+typedefpattern = re.compile(
+    r'\s*(?:,\s*(?P<access>public|private)\s*::)?\s*(?P<name>\b[\w$]+\b)\s*(?:\((?P<params>[\w,\s]*)\))?\Z', re.I)
```

Comment:
Thanks for the contribution! I am not a fortran user but I see from the reference doc that you linked to that it can also include additional access specifiers like extends and bind(c). This regex doesnt cover that, should we add support for that too or add some kind of warning message with outmess function.

Reply:
Thank you for your remark. Supporting all the Fortran standard for type definition would actually be nice target but maybe a more complex one. What I can do, is to catch attributes in the regexp, and filter them out later, emitting outmess for unsupported ones, and dealing properly with public / private.
I'm going to update the MR in that direction.

Reply:
Thanks to your suggestion, I've updated the MR. Now the regexp is not matching only 'public' or 'private' attributes but all other possible. Only access attributes are stored, others are emitting a message. I've slightly updated the proposed test to demonstrate that having also `bind(c)` as attribute is not perturbatting the detection of the access one.